### PR TITLE
YR/ Convert validation to the new format / BC103/ CIAC-10000

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-10000

## Description
Convert validation to the new format.
The validation validates that the script arguments name did not change.

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    description = "Check if the argument name has been changed."
+    rationale = "If an existing argument has been renamed, it will break backward compatibility."
+    error_message = (
+        "Possible backward compatibility break: Your updates to this file contain changes "
+        "to the names of the following existing arguments: {args}. Please undo the changes."
+    )
+    related_field = "name"  # TODO: check if this is the correct field
+    is_auto_fixable = False
+    expected_git_statuses = [GitStatuses.MODIFIED]
+    related_file_type = [RelatedFileType.YML]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py:28]:
```suggestion
    related_field = "args.name"
    expected_git_statuses = [GitStatuses.MODIFIED]
```

File: demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        results: List[ValidationResult] = []
+        for content_item in content_items:
+
+            current_args = [arg.name for arg in content_item.args]
+            old_args = [arg.name for arg in content_item.old_base_content_object.args]  # type: ignore
+            args_diff = compare_lists(sub_list=old_args, main_list=current_args)
+
+            if args_diff:
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py:38]: Maybe let's also add a check here for cases where we add a new required arg?
This is something that can happen for scripts as well.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Good job, minor comments.

File: demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py (hunk 1 above)
Comment (RosenbergYehuda) [reply demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py:28]: Done

File: demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py (hunk 2 above)
Comment (RosenbergYehuda) [reply demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py:38]: There is a validation for that IN116- will wait to hear from @JasBeilin

File: demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+class HaveTheArgsChangedValidator(BaseValidator[ContentTypes]):
+    error_code = "BC103"
+    description = "Check if the argument name has been changed."
+    rationale = "If an existing argument has been renamed, it will break backward compatibility."
+    error_message = (
+        "Possible backward compatibility break: Your updates to this file contain changes "
+        "to the names of the following existing arguments: {args}. Please undo the changes."
+    )
+    related_field = "args.name"
+    is_auto_fixable = False
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py:25]:
```suggestion
```

**Review by YuvHayun — APPROVED** [review]
Minor comments.
@JasBeilin approved from my side.

File: demisto_sdk/commands/validate/tests/BC_validators_test.py [hunk 4]
```diff
@@ -941,3 +944,56 @@ def test_IsValidToversionOnModifiedValidator_is_valid(content_items, old_content
         and result[0].message
         == "Changing the maximal supported version field `toversion` is not allowed. Please undo, or request a force merge."
     )
+
+
+def test_have_the_args_changed_validator__fails():
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BC_validators_test.py:949]:
```suggestion
def test_args_name_changed_validator__fails():
```

File: demisto_sdk/commands/validate/tests/BC_validators_test.py [hunk 5]
```diff
@@ -941,3 +944,56 @@ def test_IsValidToversionOnModifiedValidator_is_valid(content_items, old_content
...
+    old_content_items = [
+        create_script_object(paths=["args[0].name"], values=["old_arg"])
+    ]
+
+    create_old_file_pointers(modified_content_items, old_content_items)
+
+    results = HaveTheArgsChangedValidator().is_valid(modified_content_items)
+    assert (
+        " contain changes to the names of the following existing arguments: old_arg. Please undo the changes."
+        in results[0].message
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BC_validators_test.py:974]: Lets change the check to only be if our arg in the str.

File: demisto_sdk/commands/validate/tools.py [hunk 6]
```diff
@@ -147,3 +148,15 @@ def find_command(commands: List[Command], command_to_find: str) -> Optional[Comm
...
+
+
+def compare_lists(sub_list: List[str], main_list: List[str]) -> List[str]:
+    """
+    Compares two lists and returns the elements that are in the sub list but not in the main list, including duplicates.
+    Args:
+        sub_list: list of elements to compare if they are a subset of main_list
+        main_list: the list to compare against
+    Returns:
+        List the elements that appear in the sublist but not in the main list, including duplicates if they are not all present in the main list.
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tools.py:160]: Let make a better explanation of this function

File: demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Script
+
+
+class HaveTheArgsChangedValidator(BaseValidator[ContentTypes]):
+    error_code = "BC103"
+    description = "Check if the argument name has been changed."
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py:18]:
```suggestion
    description = "Check if an argument name has been changed."
```

File: demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    related_field = "args.name"
+    expected_git_statuses = [GitStatuses.MODIFIED]
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        results: List[ValidationResult] = []
+        for content_item in content_items:
+
+            current_args = [arg.name for arg in content_item.args]
+            old_args = [arg.name for arg in content_item.old_base_content_object.args]  # type: ignore
+            args_diff = compare_lists(sub_list=old_args, main_list=current_args)
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/validators/BC_validators/BC103_have_the_args_changed.py:33]: Lets put a short comment explaining why

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Great, see my comments.

**Review by JasBeilin — APPROVED** [review]
Great.

**Pull request merged**